wnfs/src/private/encrypted.rs

```diff
@@ -0,0 +1,89 @@
+use std::io::Cursor;
+
+use anyhow::Result;
+use libipld::{cbor::DagCborCodec, codec::Decode, prelude::Encode, Ipld};
+use once_cell::sync::OnceCell;
+use rand_core::RngCore;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+use crate::FsError;
+
+use super::Key;
+
+#[derive(Debug, Clone, Eq)]
+pub struct Encrypted<T> {
+    ciphertext: Vec<u8>,
+    value_cache: OnceCell<T>,
+}
+
+impl<T> Encrypted<T> {
+    pub fn from_value(value: T, key: &Key, rng: &mut impl RngCore) -> Result<Self>
```

Comment:
Do you want to add some docs here and in other public interface? The eager encryption is important to know.